src/services/engine_manager.cpp

```diff
@@ -148,7 +149,7 @@ namespace otto::services {
     });
 
     // static ScreenEnum master_last_screen = ScreenEnum::master;
-    // static ScreenEnum send_last_screen = ScreenEnum::sends;
+    static ScreenEnum send_last_screen = ScreenEnum::sends;
```

Comment:
instead of this, `UIManager` should have a stack of screens where you can set/push/pop to/from.